src/qibolab/instruments/qm.py

```diff
@@ -671,7 +724,17 @@ def add(self, pulse: Pulse):
 
         self.pulse_finish[pulse.finish].append(qmpulse)
         self.qmpulses.append(qmpulse)
-        return qmpulse
+
+    def shift(self):
+        """Shift all pulses that come after a ``BakedPulse`` a bit to avoid overlapping pulses."""
+        to_shift = collections.deque()
```

Comment:
In practice we don't need it. I doubt we will see any performance difference in our applications. I just used it because as you say it is the correct data structure for a queue (first in, first out), as the default `list` is more like a stack (last in, first out). And since it is standard library and `collections` was already imported for other use, I don't think it is adding any complexity.